wondrous-app/components/Common/CornerWidget/index.tsx

```diff
@@ -0,0 +1,52 @@
+import Snackbar from '@mui/material/Snackbar';
+import dynamic from 'next/dynamic';
+import { createContext, useContext, useMemo, useState } from 'react';
+
+import { CornerWidgetType } from './types';
```

Comment:
1. you don't need to have the types in a separate file, as you don't reuse them.
2. name it `CornerWidgetState`